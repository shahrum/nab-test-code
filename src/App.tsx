import React from "react";
import "./App.scss";
import Header from "./components/header/header.component";
import HomePage from "./pages/Home.page";

function App() {
	return (
		<div className="App">
			<Header />
			<HomePage />
		</div>
	);
}

export default App;
