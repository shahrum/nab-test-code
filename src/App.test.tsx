import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";
afterEach(cleanup);
test("renders with nab test code in the dom", () => {
	const headerText = "This is the NAB code challenge.";
	const { getByTestId } = render(<App />);
	const linkElement = screen.getByText(headerText);
	expect(linkElement).toBeInTheDocument();
	expect(getByTestId("ut-app-header")).toHaveTextContent(headerText);
});

test("snapshot", () => {
	const tree = renderer.create(<App />);
	expect(tree).toMatchSnapshot();
});
