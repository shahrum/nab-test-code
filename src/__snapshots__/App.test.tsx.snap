// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="App"
>
  <header
    aria-label="This page represent the NAB code challenge"
    data-testid="ut-app-header"
  >
    This is the NAB code challenge.
  </header>
  <div
    className="spinner"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
